kvmagent/kvmagent/plugins/vm_plugin.py

```
@@ -4796,23 +4796,55 @@ def hotplug_cpu(self, cpu_num):
+        2. QGA must be running
+        3. Guest OS must be in whitelist (ubuntu/debian) — detected via QGA
+        4. If SMBIOS manufacturer already set, skip (guest udev handles it)
+        5. Only online CPUs with logical-id >= prev_cpu_num (protect user-offlined CPUs)
+
+        Fail-safe: any exception only warns, never blocks hotplug.
         """
+        if HOST_ARCH != 'x86_64':
+            return
+
         try:
             qga = VmQga(self.domain)
```

Ruff (0.15.9)

[error] `VmQga` may be undefined, or defined from star imports

(F405)